src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -804,10 +804,24 @@ export function findOldProperty(oldTemplate, newProperty) {
  * @returns {boolean}
  */
 function shouldKeepValue(element, oldProperty, newProperty) {
+
+  // "Hidden" values are treated as a constant
```

Comment:
If we settle down on this we should backport it to C7, right?

Reply:
In C7, we don't have these checks and use the legacy, C7 specific `propertyChanged` method. 
Adjusting it in C7 would probably be a breaking change for C7 templates and I would not see it in the scope of this PR.

https://github.com/bpmn-io/bpmn-js-properties-panel/blob/a50c9166d52cfb24fe67c8fdaec609ba4dd5e3cd/src/provider/element-templates/cmd/ChangeElementTemplateHandler.js#L1231-L1293

Reply:
Perfect, thanks for clarifying. I simply forgot that we assume a more sane behavior in C8. :)